### PR TITLE
Refactor Instruction Handler Classes to Support RV32 and RV64

### PR DESCRIPTION
**Pull request by kathlenemagnus**
To reuse as much code as possible, I've removed the `rv64` subdirectory and refactored the instruction handler classes to provide a placeholder for future RV32 instruction handler functions.

### CODE REVIEW

File: core/inst_handlers/a/RvaInsts.cpp [hunk 1]
```diff
@@ -1,170 +1,211 @@
...
-            atlas::Action::createAction<&RvaInsts::sc_d_64_compute_address_handler, RvaInsts>(
-                nullptr, "sc_d", ActionTags::COMPUTE_ADDR_TAG));
-        inst_handlers.emplace(
-            "sc.w",
-            atlas::Action::createAction<&RvaInsts::sc_w_64_compute_address_handler, RvaInsts>(
-                nullptr, "sc_w", ActionTags::COMPUTE_ADDR_TAG));
-    }
-
+        static_assert(std::is_same<XLEN, RV64>::value || std::is_same<XLEN, RV32>::value);
+        if constexpr (std::is_same<XLEN, RV64>::value)
```
Comment (knute-mips) [inline core/inst_handlers/a/RvaInsts.cpp:11]: Just a thought, you can also use `if contexpr(std::is_same_v<XLEN, RV64>)`

File: core/inst_handlers/f/RvfInsts.cpp [hunk 2]
```diff
@@ -1,112 +1,137 @@
...
     {
-        inst_handlers.emplace(
-            "flw", atlas::Action::createAction<&RvfInsts::flw_64_compute_address_handler, RvfInsts>(
-                       nullptr, "flw", ActionTags::COMPUTE_ADDR_TAG));
-        inst_handlers.emplace(
-            "fsw", atlas::Action::createAction<&RvfInsts::fsw_64_compute_address_handler, RvfInsts>(
-                       nullptr, "fsw", ActionTags::COMPUTE_ADDR_TAG));
-    }
-
+        static_assert(std::is_same_v<XLEN, RV64> || std::is_same<XLEN, RV32>::value);
```
Comment (colby-nyce) [inline core/inst_handlers/f/RvfInsts.cpp:10]: You might as well use std::is_same_v in all places.

File: core/inst_handlers/f/RvfInsts.cpp [hunk 3]
```diff
@@ -1,112 +1,137 @@
...
+                atlas::Action::createAction<&RvfInsts::flw_64_compute_address_handler, RvfInsts>(
+                    nullptr, "flw", ActionTags::COMPUTE_ADDR_TAG));
+            inst_handlers.emplace(
+                "fsw",
+                atlas::Action::createAction<&RvfInsts::fsw_64_compute_address_handler, RvfInsts>(
+                    nullptr, "fsw", ActionTags::COMPUTE_ADDR_TAG));
+        }
+        else if constexpr (std::is_same_v<XLEN, RV32>)
+        {
+            sparta_assert(false, "RV32 is not supported yet!");
```
Comment (colby-nyce) [inline core/inst_handlers/f/RvfInsts.cpp:24]: Change to static_assert

**Review by colby-nyce — APPROVED** [review]

File: core/inst_handlers/f/RvfInsts.cpp (hunk 2 above)
Comment (kathlenemagnus) [reply core/inst_handlers/f/RvfInsts.cpp:10]: Bad `sed` command! I'll fix it. Good catch.

**Pull request merged**